src/core/compat/reversediff.jl

```diff
@@ -46,12 +46,10 @@ end
 @require Memoization = "6fafb56a-5788-4b4e-91ca-c0cea6611c73" @eval begin
     setrdcache(::Val{true}) = RDCache[] = true
     function emptyrdcache()
-        for k in keys(Memoization.caches)
-            if k[1] === typeof(memoized_taperesult)
-                pop!(Memoization.caches, k)
-            end
-        end
+        Memoization.empty_cache!(memoized_taperesult)
```

Comment:
This will empty any user-memoized stuff as well. The old implementation only removed the relevant parts of the memo.

Reply:
This should only remove memoized calls of `memoized_taperesult`, shouldn't it?

Reply:
BTW this fixed the cache emptying issues in https://github.com/TuringLang/Turing.jl/issues/1393 IIRC

Reply:
I also added tests that check if the cache is emptied.

Reply:
Cool, thanks!